docs/advanced/how-it-works.md

```diff
@@ -0,0 +1,175 @@
+{
+  "version": "https://schema.openattestation.com/2.0/schema.json",
+  "data": {
+    "name": "2f1a9924-bc38-455c-b39e-6420001ad67b:string:Maersk Bill of Lading",
+    "issuers": [
+      {
+        "identityProof": {
+          "type": "40caddff-5cd4-477d-adf4-48dcd0a2d761:string:DNS-TXT",
+          "location": "c15358f4-f0dc-41c8-abfb-0d030aae3233:string:imaginative-amber-ferret.sandbox.openattestation.com"
+        },
+        "name": "0de92429-f8d3-47a0-868f-154227a66f40:string:DEMO STORE",
+        "tokenRegistry": "89c1f33c-121d-4622-a561-12fb400f2f3f:string:0x8194648f40ED07F841fA357Bf52CBE8D6d7ce48D"
+      }
+    ]
+  },
+  "signature": {
+    "type": "SHA3MerkleProof",
+    "targetHash": "11d456db211d68cc8a6eac5e293422dec669b54812e4975497d7099467335987",
+    "proof": [],
+    "merkleRoot": "11d456db211d68cc8a6eac5e293422dec669b54812e4975497d7099467335987"
+  }
+}
+```
+
+Multiple interesting things happened that we will dig in:
+
+- A `data` has been created and holds the content of the file provided for wrapping with some weird-looking extra data.
+- A `signature` object has been created.
+
+### The `data` object
+
+The first step of wrapping consists of running through and transforming all the object properties provided as input:
+
+1. For each properties
+1. Generate a salt [using uuid v4](https://www.npmjs.com/package/uuid) in order to prevent [rainbow table attack](https://en.wikipedia.org/wiki/Rainbow_table).
+1. Find the type original property.
+1. Transform the original value to `<salt>:<original-type>:<original-value>`.
+
+> The shape of the input object remains untouched.
+
+### The `signature` object
+
+#### targetHash
+
+Once the `data` object has been computed we will be able to create an unique hash for the document that we will set into `targetHash`:
+
+1. List each properties' path from the `data` object and associate its value. The path follows the [flatley](https://github.com/antony/flatley) path convention.
+1. For each properties' path, compute a hash using the properties' path and value. To compute the hash we use [keccack256](https://en.wikipedia.org/wiki/SHA-3).
+1. Sort all the hashes from the previous step alphabetically and hash them all together: this will provide the `targetHash` of the document. To compute the `targetHash` we also use [keccack256](https://en.wikipedia.org/wiki/SHA-3).
+
+> The `targetHash` of a document is an unique identifier.
+
+We follow exactly the same steps to verify that a document has not been tampered.
+
+The [document store](/docs/verifiable-document/document-store) is a smart contract on the Ethereum network that records the issuance and revocation status of OpenAttestation documents. It will store hashes of wrapped documents that proves that the owner of the document store really issued the document.
+
+Imagine that you wrap a thousand of file and had to issue the `targetHash` for each of them. It would be extremely inefficient. That's where the `merkleRoot` will come handy.
```

Comment:
```suggestion
Imagine that you wrap a thousand of file and had to issue the `targetHash` for each of them. It would be extremely inefficient. That's where the `merkleRoot` will come in handy.
```

Reply:
seems a bit informal compared to the rest of the doc, but whatever